### DISCUSSION TITLE
walrus component

### DISCUSSION THREAD

**Discussion by Euraxluo** [MystenLabs/walrus-docs#83] [Ideas]
The user's portal should be able to link to more components, so I thought we well have a walrus component.

**Comment by Euraxluo** [comment]
1. we can build a object version system(maybe dont need)
2. storage components source code
3. display components on not a suins => use children object ( as a account),but can have a `namespace` ,like "🌀 Mystery Box" ,just a fontend wrapper to your package call
4. user can fork that components or use this,we publish and site it to network
5. user can use the components on there walrus site

**Comment by Euraxluo** [comment]
Tell me what you think. I'm building something like this right now. I had this idea before walruses came out,At the time I was thinking about reducing the cost to users of deploying new sites on existing networks through componentization

**Reply by Euraxluo** [reply]
you can see [0xwrapper/discover](https://github.com/0xWrapper/discover)

The relevant technical implementation paths are mentioned

**Comment by gdanezis** [comment]
Thanks for having a look at Walrus! Very curious to see  what you build, feel free to post here any progress.

**Reply by 0xWrapper** [reply]
# Discover project

### Project Background
The Discover project aims to provide a platform for developers to create front-end user interfaces for their contracts and publish these interfaces on-chain. Users can discover and use these front-end applications through this platform and bind their own domains to these applications using the on-chain domain name system. This project draws on the design of components and tools from the NEAR protocol and migrates them to the Move language environment.

### Project Objectives
1. Provide decentralized domain registration and resource management services
    - **Unique named space**: Allows users to register unique namespaces and be able to mount or access on-chain resources within the space.

2. Develop user-friendly front-end interfaces
    - **Space management**: Helps users easily manage their spaces, discover, and use on-chain applications.
    - **Contract front-end generation**: Supports AI description word generation websites to help users automatically generate and combine contract front-ends.
    - **Contract simulation execution**: Develop a contract combination simulation execution function to display execution paths and results.

3. Create a developer ecosystem
    - **Tools and APIs**: Provide development tools and APIs to help developers write front-end UIs for their smart contracts.
    - **Component marketplace**: Developers can publish their front-end components in the marketplace for other developers and users to discover and use.
    - **Build tools**: Provide a series of build tools, build platforms, and npm packages to help developers quickly develop and deploy front-end components.

4. On-chain low-cost domain system
    - **Namespace system**: Provide a namespace system that allows users to bind an independent ecosystem domain to their contract front-end.

5. Ecosystem expansion
    - **Smart applications**: Support AI description word generation websites to help users automatically generate and combine contract front-ends, develop contract combination simulation execution functions to display execution paths and results.
    - **Wallet applications**: A wallet system centered on Discover and space, helping users discover and access contract front-ends in the ecosystem and manage their space systems.


### Core functions and components
1. **Namespace/Personal Space Management**
2. **Contract Component Discovery Ecosystem**
3. **Component and Application Build Tools**
4. **Ecosystem Integration**
    - Provide API interfaces to support AI description word generation websites.
    - Support users in front-end design and contract combination through AI, provide simulation execution functions, and display execution results.
    - A wallet system centered on Discover and space, helping users discover and access contract front-ends in the ecosystem and manage their space systems.

### Reference
- [Near Protocol Developer Components](https://dev.near.org/components)
- [Near BOS Loader](https://github.com/near/bos-loader)
- [Near BOS Web Engine](https://github.com/near/bos-web-engine/deployments/Preview%20%E2%80%93%20bos-web-engine)
- [Roc Docs](https://roc-docs.near.dev/)
- [Mysten Labs Walrus Docs](https://github.com/MystenLabs/walrus-docs)

**Reply by 0xWrapper** [reply]
I have written a project vision and if we feel it makes sense to implement it, I will go further and invest zero per cent of my resources in it

**Comment by Euraxluo** [comment]
I can only build from scratch, and the good news is that I have completed the core contract build.
[0xwrapper/discover](https://github.com/0xWrapper/discover)

**Comment by Euraxluo** [comment]
i create a sip idea
https://forums.sui.io/t/create-an-interactive-interface-for-all-smart-contracts/46073
